gfe_examples/sync_test/sync_test_vertex.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import IntEnum
+import logging
+from spinn_utilities.log import FormatAdapter
+from spinn_utilities.overrides import overrides
+from pacman.model.graphs.machine import MachineVertex
+from pacman.executor.injection_decorator import inject_items
+from pacman.model.resources import ResourceContainer, ConstantSDRAM
+from spinn_front_end_common.utilities.constants import (
+    SYSTEM_BYTES_REQUIREMENT, BYTES_PER_WORD)
+from spinn_front_end_common.abstract_models import (
+    AbstractGeneratesDataSpecification, AbstractHasAssociatedBinary)
+from spinn_front_end_common.utilities.utility_objs import ExecutableType
+from spinnaker_graph_front_end.utilities.data_utils import (
+    generate_system_data_region)
+from pacman.model.graphs.application.abstract import (
+    AbstractOneAppOneMachineVertex)
+
+logger = FormatAdapter(logging.getLogger(__name__))
+
+
+SEND_PARTITION = "send"
+
+
+class DataRegions(IntEnum):
+    SYSTEM = 0
+    DATA = 1
+
+
+class SyncTestVertex(AbstractOneAppOneMachineVertex):
+
+    def __init__(self, lead, label=None, constraints=None):
+        AbstractOneAppOneMachineVertex.__init__(
+            self, SyncTestMachineVertex(lead, self, label, constraints),
+            label, None, n_atoms=1)
+
+
+class SyncTestMachineVertex(MachineVertex, AbstractHasAssociatedBinary,
+                            AbstractGeneratesDataSpecification):
```

Comment:
Is there a reason this doesn't inherit from SimulatorVertex like all the other examples do?

Reply:
I suspect it was made before SimulatorVertex was a thing, and then never updated (because it is new, it wouldn't have caused a conflict).  I guess it doesn't really do any harm, and shows that there are more than one way to do things in the GFE!

Reply:
OK, that's fair enough - there are other GFE examples like MCMC which don't use SimulatorVertex either.